aws/data_source_aws_wafv2_regex_pattern_set.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceAwsWafv2RegexPatternSet() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsWafv2RegexPatternSetRead,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"regular_expression_list": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"regex_string": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					wafv2.ScopeCloudfront,
+					wafv2.ScopeRegional,
+				}, false),
+			},
+		},
+	}
+}
+
+func dataSourceAwsWafv2RegexPatternSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	name := d.Get("name").(string)
+
+	var foundRegexPatternSet *wafv2.RegexPatternSetSummary
+	input := &wafv2.ListRegexPatternSetsInput{
+		Scope: aws.String(d.Get("scope").(string)),
+	}
+
+	for {
+		output, err := conn.ListRegexPatternSets(input)
+		if err != nil {
+			return fmt.Errorf("Error reading WAFv2 RegexPatternSets: %s", err)
+		}
+
+		for _, regexPatternSet := range output.RegexPatternSets {
+			if aws.StringValue(regexPatternSet.Name) == name {
+				foundRegexPatternSet = regexPatternSet
+				break
+			}
+		}
+
+		if output.NextMarker == nil || foundRegexPatternSet != nil {
+			break
+		}
+		input.NextMarker = output.NextMarker
+	}
+
+	if foundRegexPatternSet == nil {
+		return fmt.Errorf("WAFv2 RegexPatternSet not found for name: %s", name)
+	}
+
+	resp, err := conn.GetRegexPatternSet(&wafv2.GetRegexPatternSetInput{
+		Id:    aws.String(*foundRegexPatternSet.Id),
```

Comment:
along the lines of using the AWS SDK's conversion functions, i think ideally we'd want to use  `aws.StringValue(foundIRegexPatternSet.Id)`  though this results in a 2 conversions..this may be a case where passing the original pointer is permissible @bflad?